core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -519,10 +521,10 @@ class LogSegment private[log] (val log: FileRecords,
     }
 
     CoreUtils.tryAll(Seq(
-      () => delete(log.deleteIfExists _, "log", log.file, logIfMissing = true),
-      () => delete(offsetIndex.deleteIfExists _, "offset index", offsetIndex.file, logIfMissing = true),
-      () => delete(timeIndex.deleteIfExists _, "time index", timeIndex.file, logIfMissing = true),
-      () => delete(txnIndex.deleteIfExists _, "transaction index", txnIndex.file, logIfMissing = false)
+      () => delete(log.deleteIfExists, "log", log.file, logIfMissing = true),
+      () => delete(offsetIndex.deleteIfExists, "offset index", offsetIndex.file, logIfMissing = true),
+      () => delete(timeIndex.deleteIfExists, "time index", timeIndex.file, logIfMissing = true),
+      () => delete(txnIndex.deleteIfExists, "transaction index", txnIndex.file, logIfMissing = false)
```

Comment:
I think this will generate warnings in Scala 2.12...

Reply:
Ack. Will revert